### ISSUE TITLE
When running DistributedDataParallel, GPU0 sometimes goes to 0 utilization. 

### ISSUE THREAD

**Issue by YingleiZhang** [pytorch/pytorch#36285]
So at the moment, when GPU0 has no utilization, what is happening? The training is good and speed is fast, just want to understand a bit more about this. 
```
Thu Apr  9 00:12:57 2020       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 410.104      Driver Version: 410.104      CUDA Version: 10.0     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla V100-SXM2...  On   | 00000000:00:17.0 Off |                    0 |
| N/A   61C    P0    64W / 300W |   9174MiB / 16130MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   1  Tesla V100-SXM2...  On   | 00000000:00:18.0 Off |                    0 |
| N/A   54C    P0    77W / 300W |   9178MiB / 16130MiB |    100%      Default |
+-------------------------------+----------------------+----------------------+
|   2  Tesla V100-SXM2...  On   | 00000000:00:19.0 Off |                    0 |
| N/A   52C    P0    80W / 300W |   9178MiB / 16130MiB |    100%      Default |
+-------------------------------+----------------------+----------------------+
|   3  Tesla V100-SXM2...  On   | 00000000:00:1A.0 Off |                    0 |
| N/A   61C    P0    78W / 300W |   9116MiB / 16130MiB |    100%      Default |
+-------------------------------+----------------------+----------------------+
|   4  Tesla V100-SXM2...  On   | 00000000:00:1B.0 Off |                    0 |
| N/A   62C    P0    79W / 300W |   9158MiB / 16130MiB |    100%      Default |
+-------------------------------+----------------------+----------------------+
|   5  Tesla V100-SXM2...  On   | 00000000:00:1C.0 Off |                    0 |
| N/A   58C    P0    79W / 300W |   9198MiB / 16130MiB |    100%      Default |
+-------------------------------+----------------------+----------------------+
|   6  Tesla V100-SXM2...  On   | 00000000:00:1D.0 Off |                    0 |
| N/A   54C    P0    77W / 300W |   9198MiB / 16130MiB |    100%      Default |
+-------------------------------+----------------------+----------------------+
|   7  Tesla V100-SXM2...  On   | 00000000:00:1E.0 Off |                    0 |
| N/A   59C    P0    74W / 300W |   9198MiB / 16130MiB |    100%      Default |
+-------------------------------+----------------------+----------------------+
```

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @xush6528 @osalpekar

**Comment by SeungjunNah** [comment]
This could happen when you are synchronizing your jobs.
The process on GPU 0 may have finished its job earlier than other processes and is waiting for them to finish theirs.

**Comment by osalpekar** [comment]
Right, it seems like GPU 0 has finished computing its local gradients, and is waiting for the the other 7 workers to complete their computation before all the nodes can perform allreduce (the synchronization step). Closing the issue since this seems like normal behavior, feel free to re-open if this becomes a problem in terms of training or machine utilization going forward.
